tasks/transform_conformance/tests/babel-plugin-transform-optional-chaining/test/fixtures/oxc/keep-this/output.js

```diff
@@ -1,8 +1,21 @@
-export class Repro {
+var _repro$f, _repro$x$f, _repro$x, _repro$x$y$f, _repro$x$y, _repro$x2, _repro$x2$f, _repro$x3, _repro$x3$y$f, _repro$x4, _repro$x4$f, _repro$x$f2, _repro$x5;
+class Repro {
   test() {
-    var _this$y;
-    (_this$y = this.y) === null || _this$y === void 0
-      ? void 0
-      : _this$y.call(this);
+    var _this$f, _this$x$f, _this$x, _this$x$y$f, _this$x$y, _this$x2, _this$x2$f, _this$x3, _this$x3$y$f, _this$x4, _this$x4$f, _this$x$f2, _this$x5;
+    (_this$f = this.f) === null || _this$f === void 0 ? void 0 : _this$f.call(this);
+    (_this$x$f = (_this$x = this.x).f) === null || _this$x$f === void 0 ? void 0 : _this$x$f.call(_this$x);
+    (_this$x$y$f = (_this$x$y = this.x.y).f) === null || _this$x$y$f === void 0 ? void 0 : _this$x$y$f.call(_this$x$y);
+    (_this$x2 = this.x) === null || _this$x2 === void 0 || (_this$x2$f = _this$x2.f) === null || _this$x2$f === void 0 ? void 0 : _this$x2$f.call(_this$x2);
+    (_this$x3 = this.x) === null || _this$x3 === void 0 || (_this$x3$y$f = _this$x3.y.f) === null || _this$x3$y$f === void 0 ? void 0 : _this$x3$y$f.call(_this$x3);
+    (_this$x4 = this.x) === null || _this$x4 === void 0 || (_this$x4 = _this$x4.y) === null || _this$x4 === void 0 || (_this$x4$f = _this$x4.f) === null || _this$x4$f === void 0 ? void 0 : _this$x4$f.call(_this$x4);
+    (_this$x$f2 = (_this$x5 = this["x"]).f) === null || _this$x$f2 === void 0 ? void 0 : _this$x$f2.call(_this$x5);
   }
 }
+const repro = {};
+(_repro$f = repro.f) === null || _repro$f === void 0 ? void 0 : _repro$f.call(repro);
+(_repro$x$f = (_repro$x = repro.x).f) === null || _repro$x$f === void 0 ? void 0 : _repro$x$f.call(_repro$x);
+(_repro$x$y$f = (_repro$x$y = repro.x.y).f) === null || _repro$x$y$f === void 0 ? void 0 : _repro$x$y$f.call(_repro$x$y);
+(_repro$x2 = repro.x) === null || _repro$x2 === void 0 || (_repro$x2$f = _repro$x2.f) === null || _repro$x2$f === void 0 ? void 0 : _repro$x2$f.call(_repro$x2);
+(_repro$x3 = repro.x) === null || _repro$x3 === void 0 || (_repro$x3$y$f = _repro$x3.y.f) === null || _repro$x3$y$f === void 0 ? void 0 : _repro$x3$y$f.call(_repro$x3);
```

Comment:
Hmm, this one is still incorrect :thinking: It's picking up `repro.x` instead of `repro.x.y` as context.

- input

```js
repro.x?.y.f?.();
```

- esbuild output: https://esbuild.github.io/try/#dAAwLjI1LjAALS10YXJnZXQ9ZXMyMDE1AGNvbnN0IHJlcHJvID0ge30KcmVwcm8ueD8ueS5mPy4oKTs

```js
var _a, _b, _c;
const repro = {};
(_c = (_a = repro.x) == null ? void 0 : (_b = _a.y).f) == null ? void 0 : _c.call(_b);
```

Reply:
This case probably needs to be handled through `transform_and_join_expression`. I fixed it by copy pasting the same logic from `transform_optional_expression`. https://github.com/oxc-project/oxc/pull/9526/commits/fb55dc4350b7de3ecdd637d34f79c37a04ae16ad I'll try to clean it up.